repositories/go_repositories.bzl

```diff
@@ -39,7 +39,7 @@ def go_deps():
             name = "com_github_google_go_containerregistry",
             urls = ["https://api.github.com/repos/google/go-containerregistry/tarball/8a2841911ffee4f6892ca0083e89752fb46c48dd"],  # v0.1.4
             strip_prefix = "google-go-containerregistry-8a28419",
-            sha256 = "60b9a600affa5667bd444019a4e218b7752d8500cfa923c1ac54ce2f88f773e2",
+            sha256 = "cadb09cb5bcbe00688c73d716d1c9e774d6e4959abec4c425a1b995faf33e964",
```

Comment:
I'm not opposed to updating the sha, but is there nay concern the artifact was tampered with?

I'm not sure if this is relevant, but neither sha matches the shas of the artifacts for that version in the regular non-api project: https://github.com/bazelbuild/rules_docker/issues/1814#issuecomment-826096584

cc @alexeagle

Reply:
@alexeagle @umeshkumhar thoughts on this? Do we know why the sha suddenly changed? And why it differs from those in the repo's release page?

Reply:
@SrodriguezO  Not 100% sure, but this issue is raised with this [commit ID](https://github.com/google/go-containerregistry/commit/0976a278d3ebefb47fdbf5a2ff2b3e8a405c9e89). In this PR, you can see ko version is changed.

There are some reported issues where ko on version change has changed the tar hashes, which impacted SHA checks..More info here:
https://github.com/kubernetes/kubernetes/issues/99376

One another possibility can be that the maintainer has re-published/ updated tarball, which caused SHA change, which I don't think have happened.

I have not explored to the root cause. Let me know if that helps

Reply:
Hi,

I'm facing an issue where using rules_docker v0.16.0 results in:

```
Checksum was cadb09cb5bcbe00688c73d716d1c9e774d6e4959abec4c425a1b995faf33e964 but wanted 60b9a600affa5667bd444019a4e218b7752d8500cfa923c1ac54ce2f88f773e2
```

The root cause is explained in https://github.com/google/ko/issues/315 and https://github.com/kubernetes/kubernetes/issues/99376.

The TL;DR is that Kubernetes was using short hashes for versions, which may change in length over time. The hash is injected into the source through `git archive`.

This SHA256 change _should_ fix things for now, but will eventually break again. The real solution will be to update the dependency to a commit after https://github.com/kubernetes/kubernetes/commit/ab221b5d9f98833da2cbb5754fc2e7685bf80298.

Reply:
Further, my solution to this problem, without updating from rules_docker v0.16.0, has been to explicitly pull in github.com/google/go-containerregistry v0.4.1 via rules_go.

See https://github.com/uhthomas/6f.io/compare/8548f56df648693c0e518454cfccad0792e48eb8..3915a86d6bd9c6a376198b1d1d79979a7d4c542f